game-core/src/main/java/games/strategy/engine/pbem/NodeBbForumPoster.java

```diff
@@ -0,0 +1,171 @@
+package games.strategy.engine.pbem;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.logging.Level;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import com.github.openjson.JSONArray;
+import com.github.openjson.JSONObject;
+
+import games.strategy.engine.framework.system.HttpProxy;
+import games.strategy.net.OpenFileUtility;
+import lombok.extern.java.Log;
+
+/**
+ * Posts turn summaries to a NodeBB based forum of your choice.
+ *
+ * <p>
+ * URL format is {@code https://your.forumurl.com/api/v2/topics/<topicID>}.
+ * </p>
+ */
+@Log
+abstract class NodeBbForumPoster extends AbstractForumPoster {
+  private static final long serialVersionUID = -6128723809566917089L;
+
+  protected abstract String getForumUrl();
```

Comment:
Nit: Since this class is package-private, `protected` is just an extra token that offers nothing beyond this method simply having package-private accessibility.

Reply:
Thought about that too, I wanted to make it explicit that this is for subclasses only, probably the same reason intellij didn't complain either.
I'll change that though